engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -355,4 +361,66 @@ private static boolean isErrorLevelEnable()
     return false;
   }
 
+  /**
+   * Adds Logger Appender
+   * @param name Appender name
+   * @param className Appender class name
+   * @param args Appender args
+   */
+  public static void addAppender(String name, String className, Object args)
+  {
+    if (getAppendersNames().contains(name)) {
+      logger.warn("A logger appender with the name '{}' exists. Cannot add a new logger appender with the same name", name);
+    } else {
+      try {
+        Constructor<?> constructor = Class.forName(className).getConstructor(Object.class);
+        AppenderSkeleton appender = (AppenderSkeleton)constructor.newInstance(args);
+        appender.setName(name);
```

Comment:
The appender name can be used to change the appender properties in run-time or remove the appender. Also the appender name is used for sending of the type of the producer of the log events. And the consumer of the log events can use this name for grouping/filtering of events by their types, transformation of the event properties and storing them into a specific storage like, for instance, a table name.

It is possible to generate the unique appender name internally. But as a rule, the append name itself is an important property of the appender.